feat(jss): Updated to use JSS and migrate styles from patternfly's react styles

dlabaj opened the pull request:
Updated code to use JSS and remove the need to import the styles from react styles.  A follow up PR will be sent out to remove the console styles from patternfly repo.  This resolves issue #36 .

Note there is also support for corepack users added to this PR.

Click [here](https://react-console-popsicle.surge.sh) to see documentation preview.

nicolethoen commented on packages/module/src/components/AccessConsoles/AccessConsoles.tsx:
```diff
@@ -21,6 +19,22 @@ const isChildOfType = (child: any, type: string) => {
+    display: 'grid',
+    gridTemplateAreas:
+      `'actions-main actions-extra' + 'main main'`,
+      rowGap: 'var(--pf-v6-global--spacer--md)'
```
These global variables will not exist in v6 - they are replaced with things that start with `--pf-t-`

dlabaj replied:
done.

nicolethoen commented on packages/module/src/components/SerialConsole/XTermStyles.tsx:
```diff
@@ -0,0 +1,107 @@
+      resize: 'none'
+    },
+    '.xterm .composition-view': {
+      background: '#000',
```
There are tokens that can be used in lieu of hard coded color values

nicolethoen commented on packages/module/src/components/VncConsole/VncActions.tsx:
```diff
@@ -1,9 +1,22 @@
+  },
+  consoleActionsVnc: {
+    gridArea: 'actions-extra',
+    display: 'flex',
```
I wonder if we can use a flex layout here instead of all these flex related css overrides

dlabaj replied:
Makes sense we should add this.  Created an issue for this so that we can add this without slowing down the release as this will require some rework (https://github.com/patternfly/react-console/issues/62).

nicolethoen commented on packages/module/src/components/VncConsole/VncConsole.tsx:
```diff
@@ -10,11 +9,24 @@ import RFB from '@novnc/novnc/core/rfb';
+  },
+  consoleActionsVnc: {
+    gridArea: 'actions-extra',
+    display: 'flex',
```
I wonder if we can use a flex layout here instead of all these flex related css overrides

dlabaj replied:
Created https://github.com/patternfly/react-console/issues/62

nicolethoen reviewed (commented):
I wonder how many of these CSS overrides we could actually remove now on V6. if not removed, many will need to be updated in this PR or a follow up

wise-king-sullyman reviewed (commented):
Not sure if it's just related to the above style issues but it does look like _something_ is off with the styling in the example at least, i.e. the console loads in over the controls, and the controls look funky before the console loads.

![image](https://github.com/user-attachments/assets/e6fda5e4-1373-4231-9d2c-ab27b16ef052)


![image](https://github.com/user-attachments/assets/a2bb98bd-7f35-4c89-a402-8f13ab64f065)

fhlavac commented on packages/module/src/components/AccessConsoles/AccessConsoles.tsx:
```diff
@@ -135,9 +150,9 @@ export const AccessConsoles: React.FunctionComponent<AccessConsolesProps> = ({
   });
   return (
-    <div className={css(styles.console)}>
+    <div className={clsx(styles.console)}>
```
```suggestion
    <div className={styles.console}>
```
I think there is no need to use clsx in such cases

fhlavac commented on packages/module/src/components/AccessConsoles/AccessConsoles.tsx:
```diff
@@ -135,9 +150,9 @@ export const AccessConsoles: React.FunctionComponent<AccessConsolesProps> = ({
+    <div className={clsx(styles.console)}>
       {React.Children.toArray(children).length > 1 && (
-        <div className={css(styles.consoleActions)}>
+        <div className={clsx(styles.consoleActions)}>
```
^^

fhlavac commented on packages/module/src/components/DesktopViewer/DesktopViewer.tsx:
```diff
@@ -44,14 +43,26 @@ export interface DesktopViewerProps extends ConnectWithRemoteViewerProps {
+}: DesktopViewerProps) => {
+  const styles = useStyles();
+  return (
+  <div className={clsx(styles.consoleDesktopViewer)}>
```
^^

fhlavac replied:
```suggestion
  <div className={styles.consoleDesktopViewer}>
```

fhlavac commented on packages/module/src/components/SerialConsole/SerialConsole.tsx:
```diff
@@ -130,7 +134,7 @@ const SerialConsoleBase: React.FunctionComponent<SerialConsoleProps> = ({
         />
       )}
-      <div className={css(styles.consoleSerial)}>{terminal}</div>
+      <div className={clsx(styles.consoleSerial)}>{terminal}</div>
```
^^

fhlavac replied:
```suggestion
      <div className={styles.consoleSerial}>{terminal}</div>
```

fhlavac commented on packages/module/src/components/SerialConsole/SerialConsoleActions.tsx:
```diff
@@ -11,12 +11,22 @@ export interface SerialConsoleActionsProps extends React.HTMLProps<HTMLDivElemen
+}: SerialConsoleActionsProps) => {
+  const styles = useStyles();
+  return (
+  <div className={clsx(styles.consoleActionsSerial)}>
```
```suggestion
  <div className={styles.consoleActionsSerial}>
```

fhlavac commented on packages/module/src/components/SerialConsole/XTerm.tsx:
```diff
@@ -130,6 +134,6 @@ export const XTerm: React.FunctionComponent<XTermProps> = ({
   // ensure react never reuses this div by keying it with the terminal widget
   // Workaround for xtermjs/xterm.js#3172
-  return <div ref={ref} className="pf-v6-c-console__xterm" role="list" onFocus={onFocusIn} onBlur={onFocusOut} />;
+  return <div ref={ref} className={clsx(styles['.xterm'])} role="list" onFocus={onFocusIn} onBlur={onFocusOut} />;
```
```suggestion
  return <div ref={ref} className={styles['.xterm']} role="list" onFocus={onFocusIn} onBlur={onFocusOut} />;
```

fhlavac commented on packages/module/src/components/VncConsole/VncConsole.tsx:
```diff
@@ -235,7 +248,7 @@ export const VncConsole: React.FunctionComponent<VncConsoleProps> = ({
     <>
       {rightContent}
-      <div className={css(styles.consoleVnc)}>
+      <div className={clsx(styles.consoleVnc)}>
```
```suggestion
      <div className={styles.consoleVnc}>
```

fhlavac commented on packages/module/src/components/VncConsole/VncActions.tsx:
```diff
@@ -34,7 +48,7 @@ export const VncActions: React.FunctionComponent<VncActionProps> = ({
   };
   const toolbar = (
-    <div className={css(styles.consoleActionsVnc)}>
+    <div className={clsx(styles.consoleActionsVnc)}>
```
```suggestion
    <div className={styles.consoleActionsVnc}>
```

dlabaj commented:
> Not sure if it's just related to the above style issues but it does look like _something_ is off with the styling in the example at least, i.e. the console loads in over the controls, and the controls look funky before the console loads.
> 
> ![image](https://private-user-images.githubusercontent.com/7115053/351111902-e6fda5e4-1373-4231-9d2c-ab27b16ef052.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4NjIyNTcsIm5iZiI6MTcyMTg2MTk1NywicGF0aCI6Ii83MTE1MDUzLzM1MTExMTkwMi1lNmZkYTVlNC0xMzczLTQyMzEtOWQyYy1hYjI3YjE2ZWYwNTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjRUMjI1OTE3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzQ0NjdkZmU1MGI1OTMzNjQwNmNjZGExMTczMzQwNzhlNzNiMWVhZTIxNTQ4YjdiMDNiMDRkYjg0NDkwMjM1YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.e073ZHv5v7CR1xQvqEgv36qk7LkPkUPC3GFI1Qe8ihY)
> 
> ![image](https://private-user-images.githubusercontent.com/7115053/351111592-a2bb98bd-7f35-4c89-a402-8f13ab64f065.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4NjIyNTcsIm5iZiI6MTcyMTg2MTk1NywicGF0aCI6Ii83MTE1MDUzLzM1MTExMTU5Mi1hMmJiOThiZC03ZjM1LTRjODktYTQwMi04ZjEzYWI2NGYwNjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjRUMjI1OTE3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjkzNTA4NjNiYjkzOGIzYzZhOWIyYzZkYzlkNTI3ODQ4YTFiMjk0MzUxYWQ2YzA3NTIwZWVkMDMyOWJlZDNlZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.cYsdTLTZwe49Xfp2C6tLM07_Jun6vNgbtTP8KJTt-OM)

Fixed and updated the code.

nicolethoen commented on packages/module/src/css/xterm.css:
```diff
@@ -0,0 +1,171 @@
+
+.xterm .composition-view {
+    /* TODO: Composition position got messed up somewhere */
+    background: #000;
```
Can colors in this file be updated to tokens rather than hardcoded hex values?

dlabaj replied:
Got it.  Thanks nicole.
